low-level/sensors/impl/generic/node_hw.py

```diff
@@ -677,17 +675,17 @@ def _run_ipmitool_subcommand(self, subcommand, grep_args=None, out_file=subproce
                          % (self.ipmi_client.NAME, self.SENSOR_NAME))
             if retcode == 1:
                 if err.find(self.ipmi_client.VM_ERROR) != -1:
-                    logger.error((f"{self.SENSOR_NAME}: {self.ipmi_client.NAME}"
+                    self.request_shutdown = True
+                    raise Exception((f"{self.SENSOR_NAME}: {self.ipmi_client.NAME}"
                         f"error:: {err}\n Dependencies failed,"
                         "shutting down sensor"))
-                    self.request_shutdown = True
                 self.iem.iem_fault("IPMITOOL_ERROR")
                 if self.IPMI not in self.iem.fault_iems:
                     self.iem.fault_iems.append(self.IPMI)
             elif retcode == BASH_ILLEGAL_CMD:
-                logger.error(f"{self.SENSOR_NAME}: Required ipmitool missing \
-                    on Node. Dependencies failed, shutting down sensor")
                 self.request_shutdown = True
+                raise Exception(f"{self.SENSOR_NAME}: Required ipmitool missing \
```

Comment:
`raise with e` as described in the previous file.

Reply:
As justified above.